presto-parser/src/main/java/com/facebook/presto/sql/tree/CreateMaterializedView.java

```diff
@@ -31,6 +31,16 @@ public class CreateMaterializedView
     private final List<Property> properties;
     private final Optional<String> comment;
 
+    public CreateMaterializedView(QualifiedName name, Query query, boolean notExists, List<Property> properties, Optional<String> comment)
+    {
+        super(Optional.empty());
+        this.name = requireNonNull(name, "name is null");
+        this.query = requireNonNull(query, "query is null");
+        this.notExists = notExists;
+        this.properties = ImmutableList.copyOf(requireNonNull(properties, "properties is null"));
+        this.comment = requireNonNull(comment, "comment is null");
+    }
+
```

Comment:
Why is this needed in this PR?

Reply:
To provide a constructor without the arg Optional<NodeLocation> location. Almost all subclasses of Node has similar constructor overloading, ex., CreateTableAsSelect, and it is missing for CreateMaterializedView. This is needed by an Ast rewriter DefaultTreeRewriter, since the old NodeLocation is not valid anymore after reconstructing.